models/oneinch/fantom/oneinch_fantom_calls_transfers.sql

```diff
@@ -0,0 +1,127 @@
+{{ config( 
+    schema = 'oneinch_fantom',
+    alias = alias('calls_transfers'),
+    tags = ['dunesql'],
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash', 'start', 'transfer_trace_address']
+    )
+}}
+
+{% set project_start_date = "timestamp '2020-11-13'" %} 
+{% set lookback_days = -7 %}
+{% set transfer_selector = '0xa9059cbb' %}
+{% set transfer_from_selector = '0x23b872dd' %}
+{% set selector = 'substr(input, 1, 4)' %}
+
+with
+
+calls as (
+    select 
+        tx_hash
+        , gr_traces.start
+        , gr_traces.caller
+        , transactions.tx_from
+        , transactions.tx_success
+        , gr_traces.call_success
+        , gr_traces.call_selector
+    from (
+        select 
+            "from" as tx_from
+            , "hash" as tx_hash
+            , success as tx_success
+            , block_time from {{ source('fantom', 'transactions') }}
+        where
+        {% if is_incremental() %}
+            block_time >= date_add('day', {{lookback_days}}, now())
+        {% else %}
+            block_time >= {{ project_start_date }}
+        {% endif %}
+    ) as transactions 
+    join (
+        select 
+            tx_hash
+            , min(trace_address) as start
+            , min_by("from", trace_address) as caller
+            , min_by(success, trace_address) as call_success
+            , min_by({{selector}}, trace_address) as call_selector
+        from {{ source('fantom', 'traces') }}
+        where 
+            {% if is_incremental() %}
+                block_time >= date_add('day', {{lookback_days}}, now())
+            {% else %}
+                block_time >= {{ project_start_date }}
+            {% endif %}
+            and "to" in (
+                select distinct contract_address from {{ ref('oneinch_contract_addresses') }}
+                where project = '1inch'
+            )
+            and call_type = 'call'
+        group by tx_hash
+    ) gr_traces using(tx_hash)
+)
+
+, merged as (
+    select 
+        calls.tx_hash
+        , transfers.block_time
+        , date_trunc('day', transfers.block_time) as block_date
```

Comment:
ah, this is another oddity we found in dunesql. you need to explicitly cast this field to date in order to use in the `parition_by` column above in config. without the cast, it just takes the `block_time` timestamp data type and we can't partition on timestamp

Reply:
also, we're looking to make our partitions larger universally post-migration. maybe you can get a head start and truncate this to month, then rename to `block_month`, and partition by that? unless you use the field for something else too?

Reply:
we don't need that field tbh, so i'll just change to month and partitioning as well! also was curious why are you using daily partitions instead of monthly hah